lib/process-services/src/lib/task-list/components/task-form/task-form.component.spec.ts

```diff
@@ -768,4 +769,198 @@ describe('TaskFormComponent', () => {
+            const inputFieldTwo = fixture.debugElement.nativeElement.querySelector('#text2');
+            const inputFieldThree = fixture.debugElement.nativeElement.querySelector('#text3');
+            expect(activitFormSelector).toBeDefined();
+            expect(inputFieldOne['disabled']).toEqual(true, 'Task Form field is enabled');
+            expect(inputFieldTwo['disabled']).toEqual(true, 'Task Form field is enabled');
+            expect(inputFieldThree['disabled']).toEqual(true, 'Task Form field is enabled');
+        });
+
+        it('Should be able to disable task form fields if the task completed', async () => {
+            fixture.detectChanges();
+            getTaskDetailsSpy.and.returnValue(of(completedTaskWithFormMock));
+            component.taskId = '456';
+            component.ngOnInit();
+            fixture.detectChanges();
+            await fixture.whenStable();
+            fixture.detectChanges();
+
+            const activitFormSelector = element.querySelector('adf-form');
+            const inputFieldOne = fixture.debugElement.nativeElement.querySelector('#text1');
+            const inputFieldTwo = fixture.debugElement.nativeElement.querySelector('#text2');
+            const inputFieldThree = fixture.debugElement.nativeElement.querySelector('#text3');
+
+            expect(activitFormSelector).toBeDefined();
+            expect(inputFieldOne['disabled']).toEqual(true);
+            expect(inputFieldTwo['disabled']).toEqual(true);
+            expect(inputFieldThree['disabled']).toEqual(true);
+        });
+
+        it('Should be able to enable Form fields after claiming a task', async() => {
+            fixture.detectChanges();
+            getBpmLoggedUserSpy.and.returnValue(of(claimedTaskDetailsMock.assignee));
+            getTaskDetailsSpy.and.returnValue(of(claimedTaskDetailsMock));
+            component.taskId = '456';
+            component.ngOnInit();
+            fixture.detectChanges();
+            await fixture.whenStable();
+            fixture.detectChanges();
+
+            const activitFormSelector = element.querySelector('adf-form');
+            const inputFieldOne = fixture.debugElement.nativeElement.querySelector('#text1');
+            const inputFieldTwo = fixture.debugElement.nativeElement.querySelector('#text2');
+            const inputFieldThree = fixture.debugElement.nativeElement.querySelector('#text3');
+
+            expect(activitFormSelector).toBeDefined();
+            expect(inputFieldOne['disabled']).toEqual(false, 'Task Form field is disabled');
+            expect(inputFieldTwo['disabled']).toEqual(false, 'Task Form field is disabled');
+            expect(inputFieldThree['disabled']).toEqual(false, 'Task Form field is disabled');
+        });
+    });
+
+    describe('Task form action buttons', () => {
+
+        it('Task form action status when candidate user is not have a access to the task claimed by another candidate user', async() => {
```

Comment:
The tests should always start with `should` in the name. From what I see in this test you could use something like` Should action buttons be disabled when the user etc...`

Reply:
English grammar requires a verb after "should", never a noun.
So this would be 
"should **show** action buttons as disabled when..."
Or
"Should **disable** action buttons when..."

Reply:
I don't mind much about the grammar to be honest 😛 But I do mind about test name being meaningful and by reading it to understand the main purpose of the test

Reply:
Task Form action status does not say anything to me.

Instead 
`Should action buttons be disabled when the user etc...`
or as you proposed: `should show/disable action buttons when`

tell you important things about the test 🙂

Reply:
Updated